### ISSUE TITLE
[Enhancement] Allow setting AutomationId on DisplayAlert/DisplayActionSheet/DisplayPromptAsync

### ISSUE THREAD

**Issue by Jeanjean** [xamarin/Xamarin.Forms#13944]
## Summary
This feature has been since 2015 and was originally filed on Bugzilla [here](https://xamarin.github.io/bugzilla-archives/33/33133/bug.html) but was closed when the project was migrated to GitHub. As I believe all Elements should be accessible for automation in Xamarin.Forms I believe the request should be reopened here.

## API Changes
Add an addition overload for the DisplayAlert, DisplayActionSheet & DisplayPromptAsync methods of Page, that offer a paramater to set the AutomationId for the Element. 

e.g.

    Application.Current.MainPage.DisplayAlert("Wrong username or password. Please try again", "Error logging in", "OK", "MyAlertDialogAutomationId");

## Intended Use Case
This would allows the assertion of the proper dialog visibility in automated UITests without having to resort to nasty workarounds such as: 
iOS :: app.Query(x => x.ClassFull("_UIAlertControllerView").Descendant().Text("YOUR TEXT"))
Android :: app.Query(x => x.Class("AlertDialogLayout").Descendant().Text("YOUR TEXT"))

**Comment by Jeanjean** [comment]
Please also consider the ability to automate any buttons inside the dialog/actionsheet to allow for dismissal.

**Comment by jfversluis** [comment]
Thanks for this suggestion! As Xamarin.Forms is now in maintenance mode, this will not happen anymore for Xamarin.Forms. We're only adding bugfixes and stability fixes.

If this is still important to you, make sure to check the [.NET MAUI repo](https://github.com/dotnet/maui) and see if it's already on the roadmap. If not, feel free to open a discussion to discuss a change first or open an issue with a detailed feature request. Thanks!
